toolkit-testing/shared/src/test/scala/org/typelevel/toolkit/ToolkitCompilationTest.scala

```diff
@@ -0,0 +1,129 @@
+import munit.{CatsEffectSuite, TestOptions}
+import cats.effect.IO
+import buildinfo.BuildInfo.scalaVersion
+
+class ToolkitCompilationTest extends CatsEffectSuite {
+
+  testCompilation213("Toolkit should compile a simple Hello Cats Effect") {
+    """|import cats.effect._
+       |
+       |object Hello extends IOApp.Simple {
+       |  def run = IO.println("Hello toolkit!")
+       |}"""
+  }
+
+  testRun213("Toolkit should run a simple Hello Cats Effect") {
+    """|import cats.effect._
+       |
+       |object Hello extends IOApp.Simple {
+       |  def run = IO.println("Hello toolkit!")
+       |}"""
+  }
+
+  testCompilation3("Toolkit should compile a simple Hello Cats Effect") {
+    """|import cats.effect.*
+       |
+       |object Hello extends IOApp.Simple:
+       |  def run = IO.println("Hello toolkit!")"""
+  }
+
+  testRun3("Toolkit should run a simple Hello Cats Effect") {
+    """|import cats.effect.*
+       |
+       |object Hello extends IOApp.Simple:
+       |  def run = IO.println("Hello toolkit!")"""
+  }
+
+  testCompilation213("Toolkit should compile a script with every dependency") {
+    """|import cats.syntax.all._
+       |import cats.effect._
+       |import com.monovore.decline.effect._
+       |import fs2.data.csv.generic.semiauto._
+       |import fs2.io.file._
+       |import io.circe._
+       |import org.http4s.ember.client._
+       |
+       |object Hello extends IOApp.Simple {
+       |  def run = IO.println("Hello toolkit!")
+       |}"""
+  }
+
+  testRun213("Toolkit should run a script with every dependency") {
+    """|import cats.syntax.all._
+       |import cats.effect._
+       |import com.monovore.decline.effect._
+       |import fs2.data.csv.generic.semiauto._
+       |import fs2.io.file._
+       |import io.circe._
+       |import org.http4s.ember.client._
+       |
+       |object Hello extends IOApp.Simple {
+       |  def run = IO.println("Hello toolkit!")
+       |}"""
+  }
+
+  testCompilation3("Toolkit should compile a script with every dependency") {
+    """|import cats.syntax.all.*
+       |import cats.effect.*
+       |import com.monovore.decline.effect.*
+       |import fs2.data.csv.generic.semiauto.*
+       |import fs2.io.file.*
+       |import org.http4s.ember.client.*
+       |
+       |object Hello extends IOApp.Simple:
+       |  def run = IO.println("Hello toolkit!")"""
+  }
+
+  testRun3("Toolkit should run a script with every dependency") {
+    """|import cats.syntax.all.*
+       |import cats.effect.*
+       |import com.monovore.decline.effect.*
+       |import fs2.data.csv.generic.semiauto.*
+       |import fs2.io.file.*
+       |import org.http4s.ember.client.*
+       |
+       |object Hello extends IOApp.Simple:
+       |  def run = IO.println("Hello toolkit!")"""
+  }
+
+  def testCompilation213 = testCompilation("2.13.12")
+
+  def testCompilation3 = testCompilation("3.3.1")
+
+  def testRun213 = testCompilation("2.13.12")
+
+  def testRun3 = testCompilation("3.3.1")
```

Comment:
Can we get these version numbers out of the `BuildInfo`? By loading the `crossScalaVersions` setting or something.